sdk/eventhub/event-hubs/src/diagnostics/tracing.ts

```diff
@@ -1,136 +1,37 @@
 import { EventHubConnectionConfig } from "../eventhubConnectionConfig";
-import { OperationOptions } from "../util/operationOptions";
-import { TryAddOptions } from "../eventDataBatch";
+import { packageJsonInfo } from "../util/constants";
 
-const _createSpan = createSpanFunction({
+/**
+ * The {@link TracingClient} that is used to add tracing spans.
+ */
+export const tracingClient = createTracingClient({
   namespace: "Microsoft.EventHub",
-  packagePrefix: "Azure.EventHubs",
+  packageName: packageJsonInfo.name,
+  packageVersion: packageJsonInfo.version,
 });
 
 /**
- * Creates an EventHubs specific span, with peer.address and message_bus.destination filled out.
- * @internal
+ * Creates {@link TracingSpanOptions} from the provided data.
+ * @param eventHubConfig - The configuration object containing initial attributes to set on the span.
+ * @param spanKind - The {@link TracingSpanKind} for the newly created span.
+ * @returns a {@link TracingSpanOptions} that can be passed to a {@link TracingClient}
  */
-export function createEventHubSpan(
-  operationName: string,
-  operationOptions: OperationOptions | undefined,
-  connectionConfig: Pick<EventHubConnectionConfig, "entityPath" | "host">,
-  additionalSpanOptions?: SpanOptions
-): { span: Span; updatedOptions: OperationOptions } {
-  const { span, updatedOptions } = _createSpan(operationName, {
-    ...operationOptions,
-    tracingOptions: {
-      ...operationOptions?.tracingOptions,
-      spanOptions: {
-        // By passing spanOptions if they exist at runtime, we're backwards compatible with @azure/core-tracing@preview.13 and earlier.
-        ...(operationOptions?.tracingOptions as any)?.spanOptions,
-        ...additionalSpanOptions,
-      },
+export function toSpanOptions(
+  eventHubConfig: Pick<EventHubConnectionConfig, "entityPath" | "host">,
+  spanKind?: TracingSpanKind
+): TracingSpanOptions {
+  const spanOptions: TracingSpanOptions = {
+    spanAttributes: {
+      "message_bus.destination": eventHubConfig.entityPath,
+      "peer.address": eventHubConfig.host,
     },
-  });
-
-  span.setAttribute("message_bus.destination", connectionConfig.entityPath);
-  span.setAttribute("peer.address", connectionConfig.host);
-
-  return {
-    span,
-    updatedOptions,
   };
-}
-
-/**
- * @internal
- */
-export function createMessageSpan(
-  operationOptions: OperationOptions,
-  eventHubConfig: Pick<EventHubConnectionConfig, "entityPath" | "host">
-): ReturnType<typeof createEventHubSpan> {
-  return createEventHubSpan("message", operationOptions, eventHubConfig, {
-    kind: SpanKind.PRODUCER,
-  });
-}
-
-/**
- * Converts TryAddOptions into the modern shape (OperationOptions) when needed.
- * (this is something we can eliminate at the next major release of EH _or_ when
- * we release with the GA version of opentelemetry).
- *
- * @internal
- */
-export function convertTryAddOptionsForCompatibility(tryAddOptions: TryAddOptions): TryAddOptions {
```

Comment:
In general our policy has been that it's unnecessary to keep backwards compat for tracing options which is what this function does. `parentSpan` hasn't been a thing in over 6 months anyway, neither has spanOptions, and we rely exclusively on tracingContext throughout our code. I think it's safe to file this away but if you strongly disagree let me know